collector/zfs.go

```diff
@@ -71,6 +77,66 @@ func (c *zfsCollector) Update(ch chan<- prometheus.Metric) (err error) {
 		return err
 	}
 
+	// Zfetchstats
+	err = c.updateZfetchstats(ch)
+	switch {
+	case err == zfsNotAvailableError:
```

Comment:
Since this is checked on the first metrics set (arcstats) above, no real need to check for it in every call beyond that point.

Then you can simplify to typical `if err != nil { return err }`.